proposals/nnnn-access-level-on-imports.md

```diff
@@ -0,0 +1,321 @@
+# Access-level on import statements
+
+* Proposal: [SE-NNNN](NNNN-access-level-on-imports.md)
+* Author: [Alexis Laferrière](https://github.com/xymus)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation:
+    * [apple/swift#63912](https://github.com/apple/swift/pull/63912)
+    * [apple/swift#63948](https://github.com/apple/swift/pull/63948)
+    * [apple/swift#63974](https://github.com/apple/swift/pull/63974)
+    * [apple/swift#64014](https://github.com/apple/swift/pull/64014)
+    * [apple/swift#64033](https://github.com/apple/swift/pull/64033)
+    * [apple/swift#64047](https://github.com/apple/swift/pull/64047)
+    * [apple/swift#64693](https://github.com/apple/swift/pull/64693)
+* Upcoming Feature Flag: `InternalImports` (Enables Swift 6 behavior with imports defaulting to internal)
+* Review: ([pitch](https://forums.swift.org/t/pitch-access-level-on-import-statements/66657))
+
+## Introduction
+
+Declaring the visibility of a dependency with an access-level keyword on import statements enables enforcing which declarations can reference the imported module.
+A dependency can be marked as being visible only to the source file, module, package, or to all clients.
+This brings the familiar behavior of the access-level on declarations to dependencies and imported declarations.
+This feature can hide implementation details from clients and helps to manage dependency creep.
+
+## Motivation
+
+Good practices guides us to separate public and internal services to avoid having external clients rely on internal details. 
+Swift already offers access-levels with their respective keywords applicable to declarations and an enforcement at type-checking,
+but there is currently no equivalent official feature for dependencies.
+
+Each dependency may have a different intent;
```

Comment:
I think "dependency" is clear enough as "the thing being imported" but it might be worth just one clarifying sentence to say that when we refer to "client" in this document we always mean (3) and not (2), since (2) is of course a client of the dependency (1).

Reply:
I went with some repetition by centering everything around a library and then referring to "library dependencies" and "library clients". Hopefully it should make things more clear.